### PR TITLE
First week

### PR DESCRIPTION
**Pull request by Kondratovich**

### CODE REVIEW

File: WebAutopark.DatabaseAccess/WebAutopark.DatabaseAccess.csproj [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    <TargetFramework>net6.0</TargetFramework>
+    <ImplicitUsings>enable</ImplicitUsings>
+    <Nullable>enable</Nullable>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <Folder Include="Repositories\" />
+  </ItemGroup>
+
+  <ItemGroup>
```
Comment (kuzmiich) [inline WebAutopark.DatabaseAccess/WebAutopark.DatabaseAccess.csproj:13]: Why you don't install package dapper?

File: WebAutopark.DatabaseAccess/SQLScripts/CreateDbScript.sql [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+	[Weight] FLOAT NOT NULL,
+	[Year] INT NOT NULL,
+	[Mileage] FLOAT NOT NULL,
+	[Color] INT NOT NULL
+)
+
+CREATE TABLE [Orders]
+(
+	[OrderId] INT PRIMARY KEY NOT NULL IDENTITY(1,1),
+	[VehicleId] INT FOREIGN KEY REFERENCES Vehicles(VehicleId)
```
Comment (kuzmiich) [inline WebAutopark.DatabaseAccess/SQLScripts/CreateDbScript.sql:32]: Foreign and primary key should be on Table lvl as constraint

File: WebAutopark.DatabaseAccess/Repositories/VehicleTypeRepository.cs [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+        private readonly string QueryUpdate = "UPDATE VehicleTypes SET Name = @Name WHERE VehicleTypeId = @VehicleTypeId";
+
+        public VehicleTypeRepository(IConnectionStringProvider connectionStringProvider) :
+            base(connectionStringProvider)
+        { }
+
+        public void Create(VehicleType item) => Connection.Execute(QueryCreate, item);
+
+        public void Delete(int id) => Connection.Execute(QueryDelete, id);
```
Comment (kuzmiich) [inline WebAutopark.DatabaseAccess/Repositories/VehicleTypeRepository.cs:25]: Incorrect parameter `id` in Execute method

File: WebAutopark.DatabaseAccess/Repositories/VehicleRepository.cs [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+                                              "Color = @Color" +
+                                              "WHERE VehicleId = @VehicleId";
+
+        public VehicleRepository(IConnectionStringProvider connectionStringProvider) :
+            base(connectionStringProvider)
+        { }
+
+        public void Create(Vehicle item) => Connection.Execute(QueryCreate, item);
+
+        public void Delete(int id) => Connection.Execute(QueryDelete, id);
```
Comment (kuzmiich) [inline WebAutopark.DatabaseAccess/Repositories/VehicleRepository.cs:34]: Incorrect parameter `id` in Execute method

**Review by kuzmiich — CHANGES_REQUESTED** [review]

File: WebAutopark.DatabaseAccess/Repositories/ComponentRepository.cs [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+
+        private const string QueryUpdate = "UPDATE Components SET Name = @Name WHERE ComponentId = @ComponentId";
+
+        public ComponentRepository(IConnectionStringProvider connectionStringProvider) :
+            base(connectionStringProvider)
+        { }
+
+        public void Create(Component item) => Connection.Execute(QueryCreate, item);
+
+        public void Delete(int id) => Connection.Execute(QueryDelete, id);
```
[the remark this answers is not in the document]
Comment (Kondratovich) [reply WebAutopark.DatabaseAccess/Repositories/ComponentRepository.cs:25]: fixed

File: WebAutopark.DatabaseAccess/Repositories/OrderRepository.cs [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+
+        private const string QueryUpdate = "UPDATE Orders SET VehicleId = @VehicleId WHERE OrdersId = @OrdersId";
+
+        public OrderRepository(IConnectionStringProvider connectionStringProvider) :
+            base(connectionStringProvider)
+        { }
+
+        public void Create(Order item) => Connection.Execute(QueryCreate, item);
+
+        public void Delete(int id) => Connection.Execute(QueryDelete, id);
```
[the remark this answers is not in the document]
Comment (Kondratovich) [reply WebAutopark.DatabaseAccess/Repositories/OrderRepository.cs:25]: fixed

File: WebAutopark.DatabaseAccess/WebAutopark.DatabaseAccess.csproj (hunk 1 above)
Comment (Kondratovich) [reply WebAutopark.DatabaseAccess/WebAutopark.DatabaseAccess.csproj:13]: fixed

File: WebAutopark.DatabaseAccess/Repositories/VehicleTypeRepository.cs (hunk 3 above)
Comment (Kondratovich) [reply WebAutopark.DatabaseAccess/Repositories/VehicleTypeRepository.cs:25]: fixed

File: WebAutopark.DatabaseAccess/Repositories/VehicleRepository.cs (hunk 4 above)
Comment (Kondratovich) [reply WebAutopark.DatabaseAccess/Repositories/VehicleRepository.cs:34]: fixed

File: WebAutopark.DatabaseAccess/SQLScripts/CreateDbScript.sql (hunk 2 above)
Comment (Kondratovich) [reply WebAutopark.DatabaseAccess/SQLScripts/CreateDbScript.sql:32]: Fixed, but can you explain the benefits of declaring foreign and primary keys on table lvl as constraint?)

File: WebAutopark.Core/Entities/Order.cs [hunk 7]
```diff
@@ -0,0 +1,8 @@
+﻿namespace WebAutopark.Core.Entities
+{
+    public class Order
```
[the remark this answers is not in the document]
Comment (Kondratovich) [reply WebAutopark.Core/Entities/Order.cs:3]: image
Why? Database schema from specification has only 2 fields in Orders table

File: WebAutopark.DatabaseAccess/Repositories/OrderItemRepository.cs [hunk 8]
```diff
@@ -0,0 +1,38 @@
+﻿using Dapper;
+using WebAutopark.Core.Entities;
+using WebAutopark.Core.Interfaces;
+
+namespace WebAutopark.DatabaseAccess.Repositories
+{
+    internal class OrderItemRepository : BaseRepository, IRepository<OrderItem>
+    {
+        private readonly string QueryCreate = "INSERT INTO OrderItems (OrderId, ComponentId, Quantity) " +
```
[the remark this answers is not in the document]
Comment (Kondratovich) [reply WebAutopark.DatabaseAccess/Repositories/OrderItemRepository.cs]: I understand that Id's is better to auto-increment if possible, but in this case the fields "OrderId", "ComponentId" are foreign keys and I don't quite understand why to increment them)

[the remark this answers is not in the document]
Comment (kuzmiich) [reply WebAutopark.DatabaseAccess/Repositories/OrderItemRepository.cs]: All right.
I miss this case)

File: WebAutopark.DatabaseAccess/SQLScripts/CreateDbScript.sql (hunk 2 above)
Comment (kuzmiich) [reply WebAutopark.DatabaseAccess/SQLScripts/CreateDbScript.sql]: It's the rule of write SQL scripts.
Read MSDN.

File: WebAutopark.Core/Entities/Order.cs (hunk 7 above)
[the remark this answers is not in the document]
Comment (kuzmiich) [reply WebAutopark.Core/Entities/Order.cs:3]: Ok, don't change anything
When I created this database, I don't have db sheme.

**Review by kuzmiich — APPROVED** [review]
Good job

**Review by ArtyomKolosov2 — APPROVED** [review]

**Review by PushMeNow — APPROVED** [review]

**Pull request merged**